src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -90,23 +93,153 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
                 fileBanner,
                 cancellationToken).ConfigureAwait(false);
 
-            // Get back symbols using annotations
-            sourceDoc = newDoc.Project.Solution.GetRequiredDocument(sourceDoc.Id);
+            // get back type declaration in the newly created file
+            var destRoot = await newDoc.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var destSemanticModel = await newDoc.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            newType = destSemanticModel.GetRequiredDeclaredSymbol(destRoot.GetAnnotatedNodes(annotation).Single(), cancellationToken) as INamedTypeSymbol;
+
+            // refactor member access references in other files
+            var memberReferenceLocations = await FindMemberReferencesAsync(moveOptions.SelectedMembers, newDoc.Project.Solution, cancellationToken).ConfigureAwait(false);
+            var locationsToDoc = memberReferenceLocations.ToLookup(loc => loc.location.Document.Id);
+            var reReferencedSolution = await RefactorReferencesAsync(locationsToDoc, newDoc.Project.Solution, newType!, cancellationToken).ConfigureAwait(false);
+
+            // Possibly convert members to non-static or static if we move to/from a module
+            sourceDoc = await CorrectStaticMembersAsync(reReferencedSolution.GetRequiredDocument(sourceDoc.Id), memberNodes, newType!, cancellationToken).ConfigureAwait(false);
```

Comment:
Should the MemberPuller be in charge of performing this fixup?

Reply:
There is a set of options in MembersPuller for changing both accessibility and whether or not the member should be made non-static, but unfortunately the options are only utilized when moving to an interface, where members are made both non-static and public. 

There is no effect if we call to pull to class, as the only options that are used involve whether to make the member abstract. Changing MembersPuller to use these settings would be a fairly significant change, as we'd need to re-grab member symbols after the change (which we already are doing here because of other changes), or find the member syntaxes after they've been inserted and make the change there.

Furthermore, there is no option to change members from non-static to static, which we may want to do if we support moving from a module to a pre-existing class, in which case we need to explicitly mark all module members as shared again.